workloads/imagenet/imagenet_pytorch/workload.py

```diff
@@ -0,0 +1,235 @@
+"""Resnet workload implemented in PyTorch."""
+import contextlib
+import os
+from collections import OrderedDict
+from typing import Tuple
+
+import torch
+import torch.nn.functional as F
+from torchvision import transforms
+from torchvision.datasets import ImageNet
+from torchvision.datasets.folder import ImageFolder
+
+import spec
+import random_utils as prng
+from workloads.imagenet.workload import ImagenetWorkload
+from workloads.imagenet.imagenet_pytorch.resnet import resnet50
+from workloads.imagenet.imagenet_pytorch.utils import fast_collate, cycle
+
+
+DEVICE='cuda' if torch.cuda.is_available() else 'cpu'
+
+
+class ImagenetWorkload(ImagenetWorkload):
+
+  def __init__(self):
+    super().__init__()
+    self.dataset = 'imagenet2012'
+
+  """ data augmentation settings """
+  @property
+  def scale_ratio_range(self):
+    return (0.08, 1.0)
+
+  @property
+  def aspect_ratio_range(self):
+    return (0.75, 4.0 / 3.0)
+
+  @property
+  def center_crop_size(self):
+    return 224
+
+  @property
+  def resize_size(self):
+    return 256
+
+  @property
+  def num_train_examples(self):
+    return 1281167
+
+  @property
+  def num_eval_examples(self):
+    return 50000
+
+  @property
+  def param_shapes(self):
+    """
+    TODO: return shape tuples from model as a tree
+    """
+    raise NotImplementedError
+
+  def eval_model(
+      self,
+      params: spec.ParameterContainer,
+      model_state: spec.ModelAuxiliaryState,
+      rng: spec.RandomState,
+      data_dir: str):
+    """Run a full evaluation of the model."""
+    data_rng, model_rng = prng.split(rng, 2)
+    eval_batch_size = 128
```

Comment:
Yes, I hard-coded the same batch size used for training here, precisely to avoid OOM errors. I agree that it makes senes to generally use the same batch size for training and evaluation (or maybe allow the user control over the eval batch size, at least in "debug mode", if we choose to have one).

Reply:
From our discussion in this week's group meeting, it seems best to just use the training `batch_size` consistently for the `eval_batch_size` of all workloads.
Currently, I don't see a use in allowing the user separate control over the `eval_batch_size`, but I am happy to change my mind if someone has an example.

Reply:
Agreed. In order to make this possible, I think it's best to change the `eval_model()` [spec](https://github.com/mlcommons/algorithmic-efficiency/blob/main/spec.py#L206-L212) to also accept a `batch_size` argument, which can be passed when `eval_model()` is called in the [submission_runner](https://github.com/mlcommons/algorithmic-efficiency/blob/main/submission_runner.py#L210-L211). I created an issue (#49) for that, because all workloads will need to be adjusted and I think this PR already has enough changes (I can also address the issue later on).